metis-indexing/src/main/java/eu/europeana/indexing/common/fullbean/FieldInputUtils.java

```diff
@@ -120,6 +120,29 @@ static <T extends LiteralType> Map<String, List<String>> createLiteralMapFromStr
     return createLiteralMapFromList(Collections.singletonList(obj));
   }
 
+  /**
+   * Method that converts a ResourceOrLiteralType.class object to a multilingual map of strings
+   *
+   * @param obj The ResourceOrLiteralType object
+   * @return A Map of strings. The keys are the languages and the values is a string for
+   * the corresponding language. If the object is null, the method returns empty map.
+   */
+  static <T extends ResourceOrLiteralType> Map<String, String> createResourceOrLiteralMapSingleFromString(T obj) {
+    // Sanity check
+    if (obj == null ) {
+      return Collections.emptyMap();
+    } else {
+      final String key = Optional.of(obj).map(RESOURCE_OR_LITERAL_TYPE_KEY_GETTER)
+                                 .filter(StringUtils::isNotBlank).orElse(DEFAULT_LANG_KEY);
+      final String value = Optional.of(obj)
+                                   .map(ResourceOrLiteralType::getResource)
+                                   .map(Resource::getResource)
+                                   .orElse("");
+
+      return Map.of(key, value);
+    }
```

Comment:
I would try to not duplicate the algorithm of filling the map. You could achieve the same by building on another method in this class. For instance:
```
    final Map<String, String> result = Optional.ofNullable(createResourceOrLiteralMapFromString(obj))
        .stream().map(Map::entrySet).flatMap(Collection::stream)
        .collect(Collectors.toMap(Entry::getKey, entry -> entry.getValue().getFirst()));
    return result.isEmpty() ? null : result;
```